core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -27,35 +27,49 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
   this.setDaemon(false)
   this.logIdent = "[" + name + "]: "
   val isRunning: AtomicBoolean = new AtomicBoolean(true)
```

Comment:
Can we get rid of this field? Seems like it should be equivalent to checking `shutdownInitiated.getCount() == 0`. I think there are some references to this field directly in the code, but we can replace them with an `isRunning()` method.